### DISCUSSION TITLE
Add Secrets and Licenses to Rego Policy support

### DISCUSSION THREAD

**Discussion by kristyko** [aquasecurity/trivy#5730] [Ideas]
Currently, the ignore policy for trivy only works with vulnerabilities and misconfiguration. It would be nice to have it extended to support secrets and licenses too.

In the code `result.Secrets` and `result.Licenses` are missed when validating the results with the ignore policy: https://github.com/aquasecurity/trivy/blob/ad977a4256920d6269e84c82e4f42bbe244a2c5a/pkg/result/filter.go#L72 
 I was wondering if this is something that could be fixed?

**Comment by nikpivkin** [comment]
Hi @kristyko !

Trivy now supports [license](https://aquasecurity.github.io/trivy/v0.48/docs/scanner/license/#configuration) and [secret](https://aquasecurity.github.io/trivy/v0.48/docs/scanner/secret/#configuration) filtering.

Would you like to use Rego for more flexible filtering? Could you please share an example Rego policy for ignoring a secret or license?

**Comment by kristyko** [comment]
Hi, sorry for a delayed response!
Yes, that's what I'd like to be able to do, use Rego for filtering. I figured it shouldn’t be hard to achieve in a similar manner to how vulnerabilities or misconfigurations are already filtered.

So for example based on my needs if there’s a license to ignore I can write a Rego rule with specified values for [parameters](https://github.com/aquasecurity/trivy/blob/a3a66df0072fdcbfef7fc24a41308cc6b4ae4622/pkg/types/license.go#L7) and that should do the thing. 
Something like 
```
ignore {
	input.PkgName == “bash”
	input.Name == “GPL-3.0”
}

ignore {
	input.FilePath == <some file path>
	input.Name == <some license>
}

```
Same for secrets. Or is there something I’m missing and it’s not possible?

**Comment by kristyko** [comment]
I took the liberty of creating a PR for this https://github.com/aquasecurity/trivy/pull/6004
